### PR TITLE
Initial work to parse simd

### PR DESCRIPTION
**Pull request by certik**
Towards #2293.

### CODE REVIEW

**Comment by Thirumalai-Shaktivel** [comment]
Perfect!

File: src/lfortran/parser/tokenizer.re [hunk 1]
```diff
@@ -642,6 +643,18 @@ int Tokenizer::lex(Allocator &al, YYSTYPE &yylval, Location &loc, diag::Diagnost
...
+            pragma / newline {
+                line_num++; cur_line=cur;
+                token(yylval.string);
+                token_loc(loc);
+                if (last_token == yytokentype::TK_NEWLINE) {
+                    return yytokentype::TK_PRAGMA;
+                } else {
+                    last_token=yytokentype::TK_NEWLINE;
+                    return yytokentype::TK_EOLCOMMENT;
+                }
```
Comment (Shaikh-Ubaid) [inline src/lfortran/parser/tokenizer.re:655]: It looks like this code is executed when we have pragma followed by a newline. Hence,  I guess the last token would always be a newline. Is the else condition here, unused/redundant?

Comment (Thirumalai-Shaktivel) [reply src/lfortran/parser/tokenizer.re:655]: It's for this condition, I think:
```
print *, 123 ! LF$ something
```

Comment (Shaikh-Ubaid) [reply src/lfortran/parser/tokenizer.re:655]: Thanks! So, I think/guess the `last_token` here then means the `previous_token`?

**Pull request merged**